src/phpDocumentor/ApiReference/Factory.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.5
+ *
+ * @copyright 2010-2015 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\ApiReference;
+
+use phpDocumentor\DocumentGroup;
+use phpDocumentor\DocumentGroupDefinition as DocumentGroupDefinitionInterface;
+use phpDocumentor\DocumentGroupFactory;
+use phpDocumentor\Reflection\DocBlockFactory;
+use phpDocumentor\Reflection\Php\Factory\File\FlySystemAdapter;
+use phpDocumentor\Reflection\Php\NodesFactory;
+use phpDocumentor\Reflection\Php\ProjectFactory;
+use phpDocumentor\Reflection\Php\Factory as ProjectFactoryStrategy;
+use phpDocumentor\Reflection\PrettyPrinter;
+
+final class Factory implements DocumentGroupFactory
+{
+    /**
+     * Creates Document group using the provided definition.
+     *
+     * @param DocumentGroupDefinitionInterface $definition
+     * @return DocumentGroup
+     */
+    public function create(DocumentGroupDefinitionInterface $definition)
+    {
+        if ($definition instanceof DocumentGroupDefinition) {
```

Comment:
Why do you conditionally match here; the 'match' method prescribes that this must be a DocumentGroupDefinition; so I would expect an exception if anything else is returned. Also: this if can re-use the match method; can it not?